### ISSUE TITLE
Cannot connect to KeePassXC. Check that browser integration is enabled in KeePassXC settings.

### ISSUE THREAD

**Issue by gretzke** [keepassxreboot/keepassxc-browser#1228]
## Expected Behavior
Installing the browser extension and successfully connecting to the KeePassXC application after activating browser integration and activating the integration for a specific browser

<img width="660" alt="image" src="https://user-images.githubusercontent.com/10338391/108454487-c14cc380-726c-11eb-9a3d-85ccc1b251b4.png">

## Current Behavior
![image](https://user-images.githubusercontent.com/10338391/108454564-e3dedc80-726c-11eb-9fe1-fb76bafe15e5.png)

## Steps to Reproduce (for bugs)
1.
2.
3.
4.

## Debug info
```
Error 5: Cannot connect to KeePassXC. Check that browser integration is enabled in KeePassXC settings. keepass.js:1142:13
Connecting to native messaging host org.keepassxc.keepassxc_browser keepass.js:1059:13
Server public key: FL5fHunxSBN2cGGU3bDtYSyCYajjeqnIZsEqVEcizEA= keepass.js:638:17
Error: Could not establish connection. Receiving end does not exist.
    onReconnect moz-extension://ba820842-d55f-fa42-9b2a-5908356a9dea/background/event.js:90
event.js:93:21
Error: No content script available for this tab. keepass.js:1262:25
Uncaught (in promise) Error: keepass.keyRing[keepass.databaseHash] is undefined undefined
```

KeePassXC - 2.6.4
KeePassXC-Browser - 1.7.6
Operating system: MacIntel
Browser: Mozilla Firefox 85.0

**Comment by varjolintu** [comment]
What does the file contents in `~/Library/Application Support/Mozilla/NativeMessagingHosts` look like?

**Comment by gretzke** [comment]
<img width="523" alt="image" src="https://user-images.githubusercontent.com/10338391/108524051-ea05a500-72ce-11eb-943e-cfb8c01c1ed3.png">

**Comment by gretzke** [comment]
output of `org.keepassxc.keepassxc_browser.json`
```
{
    "allowed_extensions": [
        "keepassxc-browser@keepassxc.org"
    ],
    "description": "KeePassXC integration with native messaging support",
    "name": "org.keepassxc.keepassxc_browser",
    "path": "/Applications/KeePassXC.app/Contents/MacOS/keepassxc-proxy",
    "type": "stdio"
}
```

**Comment by varjolintu** [comment]
@gretzke Thanks. Can you try to launch `/Applications/KeePassXC.app/Contents/MacOS/keepassxc-proxy"` directly from the Terminal? Does that succeed?

**Comment by gretzke** [comment]
I can launch it. I restarted my PC and now the connection works! The window to allow access didn't pop up for me before I guess. I can't seem to reproduce the issue now. I can't rule the possibility out that the window opened on one of my other screens before, so this might be a non issue.

**Comment by varjolintu** [comment]
It's a macOS issue that the access confirmation popup can sometimes stay hidden or behind the main window. This is because macOS doesn't allow a single popup to be diplayed. So, we must always bring the main window to the foreground and then display the popup. macOS/Qt still messes up the focus sometimes.

**Comment by gretzke** [comment]
Is it somehow possible to automatically accept these requests from the specified browser? Or also the requests when trying to fill in the credentials?

**Comment by varjolintu** [comment]
Browser Integration's Advanced Settings tab has an option _Never ask before accessing credentials_ that disables this totally. It's disabled by default because it's recommended to have a user confirmation before accessing anything from the database.

**Comment by gretzke** [comment]
I totally understand this from a security standpoint. Is the access still protected by the URL even if enabling _Never ask before accessing credentials_. E.g., could a phishing website request access for some credentials or would this be blocked?

**Comment by varjolintu** [comment]
@gretzke Your credentials could be potentially accessed IF someone could know your Client ID you've saved in your browser + the key pair you are using without the confirmation check. The possibility for that is still quite small, and would need a malicious script or malware application. Currently we are also guarding the usernames in the web page inside a closed Shadow DOM so the web page or its scripts cannot access to that data either. Your passwords are never visible, except when you actually fill them to the password field, obviously.

So in short, a website could try to access your credentials but still cannot view them.
